src/databricks_ai_bridge/genie.py

```diff
@@ -144,12 +148,20 @@ def poll_result():
                             "content"
                         ]
                         return GenieResponse(result=text_content)
-                elif resp["status"] in ["FAILED", "CANCELLED", "QUERY_RESULT_EXPIRED"]:
-                    logging.debug(f"Genie query {resp['status'].lower()}.")
-                    return GenieResponse(result=None)
+                elif resp["status"] in ["CANCELLED", "QUERY_RESULT_EXPIRED"]:
+                    return GenieResponse(result=f"Genie query {resp['status'].lower()}.")
+                elif resp["status"] == "FAILED":
+                    return GenieResponse(
+                        result=f"Genie query failed with error: {resp.get('error', 'Unknown error')}"
+                    )
                 else:
                     logging.debug(f"Waiting...: {resp['status']}")
                     time.sleep(5)
+            return GenieResponse(
```

Comment:
Nit: change this message to `Genie result timed out ...` so we can differentiate it from the above time out message for debugging